EIPS/eip-3450.md

```diff
@@ -0,0 +1,151 @@
+Mnemonics are a great UX (and security) improvement over hex secrets. But, they are still a single point of failure.
+
+Shamir's Secret Sharing Scheme is great for sharing secrets so that there is no single point of failure. But, there is no standard implementation of Sharmir's, putting recovery at risk.
+
+In this EIP, we combine mnemonics with Sharmir's Secret Sharing Scheme and propose a standardized algorithm to make storing keys easier and more secure.
+
+## Specification
+
+For a general overview of Shamir's Secret Sharing Scheme see:
+
+- [Wikipedia: Shamir's Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_Secret_Sharing)
+- [SLIP-0039: Shamir's Secret-Sharing for Mnemonic Codes](https://github.com/satoshilabs/slips/blob/master/slip-0039.md#shamirs-secret-sharing)
+
+There are two operations: Splitting a BIP-39 mnemonic into shares and recovering the original mnemonic with a subset of the shares.
+
+We use the [standard BIP-39 english word list](https://github.com/bitcoin/bips/blob/master/bip-0039/english.txt) with 2048 words, in a 0-indexed array.
+
+Accordingly, we use GF(2048) (or GF(2<sup>11</sup>)) with primitive polynomial x<sup>11</sup> + x<sup>2</sup> + 1 (or 5) for all arithmetic.
+
+### Definition: Share
+
+A share represents a point on the curve described by the underlying polynomial used to split the secret. It includes two pieces of data:
+
+- An id: the _x_ value of the share
+- A BIP-39 mnemonic: the _y_ value of the share represented by a mnemonic
+
+### Polynomial Interpolation
+
+Interpolation is used to recover the mnemonic. See [Recover Mnemonic](#recover-mnemonic) below.
+
+This follows the [SLIP-0039 specification for polynomial interpolation](https://github.com/satoshilabs/slips/blob/master/slip-0039.md#polynomial-interpolation) closely.
+
+Given a set of _K_ points (_x<sub>i</sub>_, _y<sub>i</sub>_), 1 ≤ _i_ ≤ _K_, such that no two _x<sub>i</sub>_ values equal,
+there exists a polynomial that assumes the value _y<sub>i</sub>_ at each point _x<sub>i</sub>_. The polynomial of lowest degree
+that satisfies these conditions is uniquely determined and can be obtained using the Lagrange interpolation formula given below.
+
+Shamir's secret sharing scheme is applied separately to each of the _n_ words (or, the integer representation of each word) of the shared mnemonic, so we work with
+_y<sub>i</sub>_ as a vector of _n_ values, where _y<sub>i</sub>_[_l_] = _f<sub>l</sub>_(_x<sub>i</sub>_), 1 ≤ _l_ ≤ _n_,
+and _f<sub>j</sub>_ is the polynomial in the _j_-th instance of the scheme.
+
+Interpolate(_x_, {(_x<sub>i</sub>_, _y<sub>i</sub>_), 1 ≤ _i_ ≤ _K_})
+
+Input: the desired index _x_, a set of index/value-vector pairs {(_x<sub>i</sub>_, _y<sub>i</sub>_), 1 ≤ _i_ ≤ _K_} ⊆ GF(2048) × GF(2048)<sup>n</sup>
+
+Output: the value-vector (_f<sub>1</sub>_(_x_), ... , _f<sub>n</sub>_(_x_))
+
+![$f_l(x) = \sum_{i=1}^{K} y_i[l] \prod_{\underset{j \neq i}{j=1}}^{K} \frac{x - x_j}{x_i - x_j}$](../assets/eip-3450/lagrange.gif)
```

Comment:
While I don't disagree in most cases, mathematical formulas like this can be difficult to write out with more descriptive variables.

This particular formula is a form of Lagrange, and is relatively common. So, I feel like the mathematical notation here is reasonable.